In [1]:
import numpy as np

**O que são arrays numpy?**

In [6]:
# vamos usar uma função do numpy para ler nossos arquivos
km = np.loadtxt('carros-km.txt')
km

array([4.44100e+04, 5.71200e+03, 3.71230e+04, 0.00000e+00, 2.57570e+04,
       1.07280e+04, 0.00000e+00, 7.75990e+04, 9.91970e+04, 3.79780e+04,
       1.28590e+04, 8.05200e+03, 8.97730e+04, 0.00000e+00, 4.14570e+04,
       1.15607e+05, 4.64490e+04, 0.00000e+00, 3.70860e+04, 1.51730e+04,
       1.01193e+05, 0.00000e+00, 9.80790e+04, 1.02959e+05, 0.00000e+00,
       0.00000e+00, 5.79500e+03, 0.00000e+00, 5.88480e+04, 9.43810e+04,
       3.01630e+04, 5.33320e+04, 1.77200e+04, 3.38080e+04, 9.06840e+04,
       4.39750e+04, 0.00000e+00, 0.00000e+00, 5.52600e+03, 0.00000e+00,
       9.34150e+04, 4.07620e+04, 0.00000e+00, 8.63020e+04, 0.00000e+00,
       9.75500e+03, 6.99450e+04, 2.39500e+03, 0.00000e+00, 8.03490e+04,
       8.55540e+04, 5.04960e+04, 6.77160e+04, 9.39470e+04, 3.53450e+04,
       8.10070e+04, 1.19513e+05, 0.00000e+00, 0.00000e+00, 0.00000e+00,
       0.00000e+00, 0.00000e+00, 1.18895e+05, 4.85090e+04, 1.00912e+05,
       9.56490e+04, 0.00000e+00, 9.04950e+04, 0.00000e+00, 2.913

In [5]:
# faremos a mesma coisa com o arquivo contendo os anos dos veículos
# veja que o argumento dtype nos diz qual o tipo de dado deverá ser convertido
anos = np.loadtxt('carros-anos.txt', dtype=int)
anos

array([2003, 1991, 1990, 2019, 2006, 2012, 2019, 2009, 2010, 2011, 2002,
       2007, 2001, 2019, 2009, 2016, 2012, 2019, 2016, 2017, 2002, 2019,
       1994, 1991, 2019, 2019, 2016, 2019, 2002, 1991, 1997, 1995, 2018,
       1991, 2014, 1995, 2019, 2019, 2009, 2019, 2013, 2003, 2019, 2002,
       2019, 2002, 2004, 2001, 2019, 2003, 2009, 2006, 2000, 2006, 2010,
       1994, 2007, 2019, 2019, 2019, 2019, 2019, 2017, 2011, 2001, 1996,
       2019, 1994, 2019, 2007, 1994, 2007, 1997, 2006, 2013, 1994, 2008,
       2008, 2019, 2019, 1999, 2007, 1993, 1995, 2019, 2019, 2016, 2011,
       2019, 2002, 2013, 2017, 2016, 2008, 2018, 2003, 2003, 2004, 2010,
       2019, 2006, 2001, 2019, 2015, 2018, 2017, 1992, 2001, 1993, 1995,
       2003, 2014, 2008, 2013, 2019, 2017, 2019, 2019, 2006, 2015, 2007,
       2005, 2017, 1998, 2019, 1999, 2019, 2009, 2003, 1994, 2019, 1994,
       2019, 2015, 1996, 1997, 1993, 2019, 1992, 1995, 2019, 1990, 2019,
       2019, 1991, 1990, 2012, 2007, 2013, 2000, 20

In [10]:
# agora vamos calcular quando kilometros por ano cada carro andou
# para fazer isso basta dividirmos a kilometragem pela idade do carro
km_media = km / (2021 - anos)
km_media

array([2.46722222e+03, 1.90400000e+02, 1.19751613e+03, 0.00000000e+00,
       1.71713333e+03, 1.19200000e+03, 0.00000000e+00, 6.46658333e+03,
       9.01790909e+03, 3.79780000e+03, 6.76789474e+02, 5.75142857e+02,
       4.48865000e+03, 0.00000000e+00, 3.45475000e+03, 2.31214000e+04,
       5.16100000e+03, 0.00000000e+00, 7.41720000e+03, 3.79325000e+03,
       5.32594737e+03, 0.00000000e+00, 3.63255556e+03, 3.43196667e+03,
       0.00000000e+00, 0.00000000e+00, 1.15900000e+03, 0.00000000e+00,
       3.09726316e+03, 3.14603333e+03, 1.25679167e+03, 2.05123077e+03,
       5.90666667e+03, 1.12693333e+03, 1.29548571e+04, 1.69134615e+03,
       0.00000000e+00, 0.00000000e+00, 4.60500000e+02, 0.00000000e+00,
       1.16768750e+04, 2.26455556e+03, 0.00000000e+00, 4.54221053e+03,
       0.00000000e+00, 5.13421053e+02, 4.11441176e+03, 1.19750000e+02,
       0.00000000e+00, 4.46383333e+03, 7.12950000e+03, 3.36640000e+03,
       3.22457143e+03, 6.26313333e+03, 3.21318182e+03, 3.00025926e+03,
      

In [15]:
# verificando o tipo de dados do array que criamos
type(km_media)

numpy.ndarray

**Operações com variáveis**

In [39]:
ano_atual = 2021
ano_fabricacao = 2003
km_total = 44410
km_media = km_total / (ano_atual - ano_fabricacao)
km_media

2467.222222222222

>Ou seja, este carro fez em média 2467 km por ano

In [41]:
# se quisermos calcular qual o total de km desse carro no próximo ano basta somarmos o total à média
# usar o += é equivalente a somar a variável com o valor que vem após o símbolo de igual
km_total += km_media

km_total

49344.44444444444

**Declaração múltipla**

In [42]:
# vamos declarar as mesmas variáveis acimas, mas agora de uma maneira diferente
ano_atual, ano_fabricacao, km_total = 2021, 2003, 44410

In [46]:
valor = 2
quadrado = valor ** 2
quadrado **= 2
quadrado

16